engine/src/main/java/org/hibernate/search/query/engine/spi/EntityInfo.java

```diff
@@ -23,9 +22,11 @@
 
 	Object[] getProjection();
 
-	List<Integer> getIndexesOfThis();
+	Iterable<Integer> getIndexesOfThis();
```

Comment:
The only thing you can do is to iterate over it. Not even a simple empty/size check is possible. If you are afraid about mutation, wrapping into a unmodifiable collection is the better option imo.